fe/fe-core/src/test/java/com/starrocks/alter/SchemaChangeHandlerWithMVTest.java

```diff
@@ -0,0 +1,434 @@
+        MaterializedIndex baseIndex = partition.getBaseIndex();
+        List<Tablet> tablets = baseIndex.getTablets();
+        for (Tablet tablet : tablets) {
+            List<Replica> replicas = ((LocalTablet) tablet).getImmutableReplicas();
+            for (Replica replica : replicas) {
+                replica.updateVersionInfo(version, -1, version);
+            }
+        }
+    }
+
+    private void waitAlterJobDone(Map<Long, AlterJobV2> alterJobs) throws Exception {
+        for (AlterJobV2 alterJobV2 : alterJobs.values()) {
+            while (!alterJobV2.getJobState().isFinalState()) {
+                LOG.info("alter job {} is running. state: {}", alterJobV2.getJobId(), alterJobV2.getJobState());
+                Thread.sleep(500);
+            }
+            LOG.info("alter job {} is done. state: {}", alterJobV2.getJobId(), alterJobV2.getJobState());
+            Assert.assertEquals(AlterJobV2.JobState.FINISHED, alterJobV2.getJobState());
+
+            OlapTable tbl = (OlapTable) starRocksAssert.getTable("test", alterJobV2.tableName);
+            while (tbl.getState() != OlapTableState.NORMAL) {
+                Thread.sleep(500);
+            }
+        }
+    }
+    private void checkModifyColumnsWithMaterializedViews(StarRocksAssert starRocksAssert,
+                                                         String mv, String alterColumn) throws Exception {
+        checkModifyColumnsWithMaterializedViews(starRocksAssert, mv, alterColumn, true);
+    }
+
+    private void checkModifyColumnsWithMaterializedViews(StarRocksAssert starRocksAssert,
+                                                         String mv, String alterColumn,
+                                                         boolean isDropMV) throws Exception {
+        String mvName = starRocksAssert.getMVName(mv);
+        try {
+            starRocksAssert.withRefreshedMaterializedView(mv);
+
+            AlterTableStmt dropValColStm = (AlterTableStmt) parseAndAnalyzeStmt(alterColumn);
+            GlobalStateMgr.getCurrentState().getAlterJobMgr().processAlterTable(dropValColStm);
+
+            Map<Long, AlterJobV2> alterJobs = GlobalStateMgr.getCurrentState().getSchemaChangeHandler().getAlterJobsV2();
+            waitAlterJobDone(alterJobs);
+        } finally {
+            if (isDropMV) {
+                try {
+                    starRocksAssert.dropMaterializedView(mvName);
+                } catch (Exception e) {
+                    // ignore.
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testModifyColumnsWithNoAggregateRollup1() {
+        try {
+            checkModifyColumnsWithMaterializedViews(starRocksAssert,
+                    "create materialized view mv1 as select error_code, error_msg, " +
+                            "timestamp from sc_dup3",
+                    "alter table sc_dup3 drop column op_id");
+            jobSize++;
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
remove this?

Reply:
it's for better debugging? no need ignore?